.github/workflows/docker-build-and-test.yml

```diff
@@ -36,8 +36,16 @@ jobs:
       - name: Install deps
         run: pip install -r ./.dagger-ci/daggerci/requirements.txt
       - uses: actions/checkout@v3
-      - name: Run Dagger pipeline
-        run: python .dagger-ci/daggerci/main.py -d ${{ matrix.dockerfile }} --publish
+      - name: Run dagger pipeline
+        run: |
+          if [[ "${GITHUB_EVENT_NAME}" == 'release' ]] || [[ "${GITHUB_REF}" == *'main' ]]; then
```

Comment:
I'm usually a fan of splitting jobs that are only supposed to run on a certain branch into a separate yaml and adapt the `on:` stanza but given that would cause a lot of duplication here it should be fine to have a condition here. And using the yaml builtin `if` (https://docs.github.com/en/actions/learn-github-actions/expressions#example-expression-in-an-if-conditional) is a bit too cursed as well.

Reply:
Yeah, I had the exact same dilema. I believe GitLab provide such option, that you can just reference another job, and change a thing or two very efficiently. But I have not found similar feature in GitHub :(

As for the yaml `if`, if does not have `else`, which would make things rather ugly, especially if we later decide to add more conditions (like schedule event for example).

This is not my proudest work, but should do the trick.